compiler/rustc_typeck/src/coherence/orphan.rs

```diff
@@ -330,6 +344,24 @@ fn emit_orphan_check_error<'tcx>(
     })
 }
 
+fn emit_orphan_check_for_impl_on_types<'tcx, T: rustc_errors::EmissionGuarantee>(
+    full_impl_span: Span,
+    self_ty_span: Span,
+    ptr_ty: &ty::TypeAndMut<'_>,
+    diag: &mut DiagnosticBuilder<'tcx, T>,
```

Comment:
We can actually just do this apparently (and remove `T` parameter)

```suggestion
    diag: &mut Diagnostic,
```